src/openms/include/OpenMS/ANALYSIS/OPENSWATH/PeakIntegrator.h

```diff
@@ -624,6 +720,250 @@ namespace OpenMS
+      with respect to `mu` (the mean).
+
+      Needed by the gradient descent algorithm.
+
+      @param[in] xs Positions
+      @param[in] ys Intensities
+      @param[in] h Amplitude
+      @param[in] mu Mean
+      @param[in] sigma Standard deviation
+      @param[in] tau Exponent relaxation time
+
+      @return The computed cost
+    */
+    double E_wrt_mu(
+      const std::vector<double>& xs,
+      const std::vector<double>& ys,
+      const double h,
+      const double mu,
+      const double sigma,
+      const double tau
+    ) const;
+
+    /**
+      @brief Compute the cost given by the partial derivative of the loss function E,
+      with respect to `sigma` (the standard deviation).
+
+      Needed by the gradient descent algorithm.
+
+      @param[in] xs Positions
+      @param[in] ys Intensities
+      @param[in] h Amplitude
+      @param[in] mu Mean
+      @param[in] sigma Standard deviation
+      @param[in] tau Exponent relaxation time
+
+      @return The computed cost
+    */
+    double E_wrt_sigma(
+      const std::vector<double>& xs,
+      const std::vector<double>& ys,
+      const double h,
+      const double mu,
+      const double sigma,
+      const double tau
+    ) const;
+
+    /**
+      @brief Compute the cost given by the partial derivative of the loss function E,
+      with respect to `tau` (the exponent relaxation time).
+
+      Needed by the gradient descent algorithm.
+
+      @param[in] xs Positions
+      @param[in] ys Intensities
+      @param[in] h Amplitude
+      @param[in] mu Mean
+      @param[in] sigma Standard deviation
+      @param[in] tau Exponent relaxation time
+
+      @return The computed cost
+    */
+    double E_wrt_tau(
+      const std::vector<double>& xs,
+      const std::vector<double>& ys,
+      const double h,
+      const double mu,
+      const double sigma,
+      const double tau
+    ) const;
+
+    /**
+      @brief Compute EMG's z parameter
+
+      The value of z decides which formula is to be used during EMG function computation.
+      The interested ranges for z are:
+      (-inf, 0), [0, 6.71e7], (6.71e7, +inf)
+
+      Paper:
+      Kalambet, Y.; Kozmin, Y.; Mikhailova, K.; Nagaev, I.; Tikhonov, P. (2011). "Reconstruction of chromatographic peaks using the exponentially modified Gaussian function". Journal of Chemometrics. 25 (7): 352.
+
+      @param[in] x Position
+      @param[in] mu Mean
+      @param[in] sigma Standard deviation
+      @param[in] tau Exponent relaxation time
+
+      @return The computed parameter z
+    */
+    double compute_z(
+      const double x,
+      const double mu,
+      const double sigma,
+      const double tau
+    ) const;
+
+    /**
+      @brief Compute the EMG function on a set of points.
+
+      If the argument `compute_additional_points` is `true`, the algorithm will
+      detect which side of the peak is cutoff and add points to it.
```

Comment:
We should probably be explicit in our terminology.  Somewhere in this method or at the top level method we need to define a cutoff and saturated peak.  

cutoff peak:  The intensities of the left and right baselines are not equal
saturated peak: The maximum intensity of the peak is lower than expected due to saturation of the detector